js/steps.js

```diff
@@ -0,0 +1,143 @@
+class Steps {
+  constructor({ selector = null, steps = null, defaultValidator = false, validateInOrder = true, navigateOnClick = !validateInOrder } = {}) {
+    this.rootElement = document.querySelector(selector);
+    this.steps = steps;
+    this.childElements = [];
+    this.navigateOnClick = navigateOnClick;
+    this.validateInOrder = validateInOrder;
+    this.defaultValidator = defaultValidator;
+    this.createAllHTMLElements();
+    this.changeActiveStep(0);
+  }
+  changeActiveStep(stepIndex) {
+    this.activeStep = stepIndex;
+    this.childElements.map(childElement => childElement.stepContainer.classList.remove("active"));
+    this.childElements[stepIndex].stepContainer.classList.add("active");
+    this.changeProgressBarWidth(this.childElements[stepIndex].stepContainer);
+  }
+  completeStep(isValid) {
+    let classValue = isValid ? "valid" : "invalid";
+    this.CurrentElement.stepContainer.classList.remove("valid");
+    this.CurrentElement.stepContainer.classList.remove("invalid");
+    this.CurrentElement.stepContainer.classList.add(classValue);
+    this.CurrentElement.isValid = isValid;
+  }
+  validateStep(isValid) {
+    if (isValid != null || this.validateInOrder) {
+      isValid = isValid === null ? this.defaultValidator : isValid;
```

Comment:
try to not reassign a function parameter, better to create a new variable

Reply:
The parameter is reassigned only when not given or null. Should I create a new variable in this case?